src/System.Net.HttpListener/tests/AuthenticationTests.cs

```diff
@@ -47,6 +47,23 @@ public AuthenticationTests()
             }
         }
 
+        [ConditionalTheory(nameof(Helpers) + "." + nameof(Helpers.IsWindowsImplementation))] // Managed implementation connects successfully.
```

Comment:
That makes sense. I couldn't locate a bug similar to the described issue so I've opened https://github.com/dotnet/corefx/issues/20840.
Added the issue within the code as well.